### ISSUE TITLE
RedundantCondition not issued for isset().

### ISSUE THREAD

**Issue by rzvc** [vimeo/psalm#7965]
```php
/** @param array<string, string> $arr */
function test(array $arr) : void
{
	if (!isset($arr['asdf']))
		return;
    
	if (isset($arr['asdf']))
		echo $arr['asdf'];
}
```
https://psalm.dev/r/fa711a0367

Was expecting a RedundantCondition, because isset has already been asserted. [Removing both checks](https://psalm.dev/r/fe90ecb9b9) yields [PossiblyUndefinedStringArrayOffset](https://psalm.dev/216), and leaving only the first one in [doesn't generate any issue](https://psalm.dev/r/a99b5b5308), so Psalm is aware that isset has been asserted already.

**Comment by AndrolGenhald** [comment]
I think removing this resulted in a bunch of false positives, so it might not be as easy as you'd think. It's been a while though, so idk.
